nodes/text/export_gcode.py

```diff
@@ -0,0 +1,298 @@
+
+from sverchok.utils.sv_text_io_common import (
+    FAIL_COLOR, READY_COLOR, TEXT_IO_CALLBACK,
+    get_socket_type,
+    new_output_socket,
+    name_dict,
+    text_modes
+)
+
+def convert_to_text(list):
+    while True:
+        if type(list) is str: break
+        elif type(list) in (tuple, list):
+            try:
+                list = '\n'.join(list)
+                break
+            except: list = list[0]
+        else: break
+    return list
+
+class SvExportGcodeNode(bpy.types.Node, SverchCustomTreeNode):
+    """
+    Triggers: Export gcode from vertices position
+    Tooltip: Generate a gcode file from a list of vertices
+    """
+    bl_idname = 'SvExportGcodeNode'
+    bl_label = 'Export Gcode'
+    bl_icon = 'COPYDOWN'
+
+    last_e = FloatProperty(name="Pull", default=5.0, min=0, soft_max=10)
+    path_length = FloatProperty(name="Pull", default=5.0, min=0, soft_max=10)
+
+    folder = StringProperty(name="File", default="", subtype='FILE_PATH')
+    pull = FloatProperty(name="Pull", default=5.0, min=0, soft_max=10)
+    push = FloatProperty(name="Push", default=5.0, min=0, soft_max=10)
+    dz = FloatProperty(name="dz", default=2.0, min=0, soft_max=20)
+    flow_mult = FloatProperty(name="Flow Mult", default=1.0, min=0, soft_max=3)
+    feed = IntProperty(name="Feed Rate (F)", default=1000, min=0, soft_max=20000)
+    feed_horizontal = IntProperty(name="Feed Horizontal", default=2000, min=0, soft_max=20000)
+    feed_vertical = IntProperty(name="Feed Vertical", default=500, min=0, soft_max=20000)
+    feed = IntProperty(name="Feed Rate (F)", default=1000, min=0, soft_max=20000)
+    esteps = FloatProperty(name="E Steps/Unit", default=5, min=0, soft_max=100)
+    start_code = StringProperty(name="Start", default='')
+    end_code = StringProperty(name="End", default='')
+    auto_sort = BoolProperty(name="Auto Sort", default=True)
+    close_all = BoolProperty(name="Close Shapes", default=False)
+    nozzle = FloatProperty(name="Nozzle", default=0.4, min=0, soft_max=10)
+    layer_height = FloatProperty(name="Layer Height", default=0.1, min=0, soft_max=10)
+    filament = FloatProperty(name="Filament (\u03A6)", default=1.75, min=0, soft_max=120)
+
+    gcode_mode = EnumProperty(items=[
+            ("CONT", "Continuous", ""),
+            ("RETR", "Retraction", "")
+        ], default='CONT', name="Mode")
+
+    def sv_init(self, context):
+        self.inputs.new('StringsSocket', 'Layer Height', 'Layer Height').prop_name = 'layer_height'
+        self.inputs.new('StringsSocket', 'Flow Mult', 'Flow Mult').prop_name = 'flow_mult'
+        self.inputs.new('VerticesSocket', 'Vertices', 'Vertices')
+
+        self.outputs.new('StringsSocket', 'Info', 'Info')
+        self.outputs.new('VerticesSocket', 'Vertices', 'Vertices')
+        self.outputs.new('StringsSocket', 'Printed Edges', 'Printed Edges')
+        self.outputs.new('StringsSocket', 'Travel Edges', 'Travel Edges')
```

Comment:
please drop all duplicates  (in 2.8 we'd have to do that anyway, or supply a pointless identifier kw
```diff
- self.outputs.new('StringsSocket', 'Travel Edges', 'Travel Edges')
+ self.outputs.new('StringsSocket', 'Travel Edges',)
```

Reply:
Done, thanks!